### ISSUE TITLE
Cannot add calendar event with location

### ISSUE THREAD

**Issue by Digitwellsolutions** [nextcloud/calendar#4335]
### Steps to reproduce

From Browser,
Go to Calendar on NextCloud
Create new Event.
Set Start and end times
Fill in a Location
Hit save
A spinning wheel appears and comtinues.

If the page is Manually refreshed, the event has not been created.
Checking the log, there is no error in the log.

From Thunderbird,
Create a calendar event with a location
The following appears in the Thunderbird error log
PUT   https://nextcloud.redacted.xxx/index.php/apps/dashboard/
[HTTP/2 405 Method Not Allowed 290ms]

Calendar: CalDAV: Unexpected status adding item to Digitwell Solutions: 405
BEGIN:VCALENDAR
PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN
VERSION:2.0
BEGIN:VTIMEZONE
TZID:Europe/London
BEGIN:DAYLIGHT
TZOFFSETFROM:+0000
TZOFFSETTO:+0100
TZNAME:BST
DTSTART:19700329T010000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3
END:DAYLIGHT
BEGIN:STANDARD
TZOFFSETFROM:+0100
TZOFFSETTO:+0000
TZNAME:GMT
DTSTART:19701025T020000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
CREATED:20220630T094432Z
LAST-MODIFIED:20220630T094519Z
DTSTAMP:20220630T094519Z
UID:da3c7599-3035-4fce-a2c0-a73da626668f
SUMMARY:trail
DTSTART;TZID=Europe/London:20220630T110000
DTEND;TZID=Europe/London:20220630T120000
TRANSP:OPAQUE
LOCATION:around here
BEGIN:VALARM
ACTION:DISPLAY
TRIGGER;VALUE=DURATION:-PT15M
DESCRIPTION:Default Mozilla Description
END:VALARM
END:VEVENT
END:VCALENDAR
CalDavCalendar.jsm:654

### Expected behavior

The calendar event should be added

### Actual behaviour

The calendar event is not added

### Calendar app version

3.4.1

### CalDAV-clients used

Thunderbird lightning

### Browser

Firefox 102

### Client operating system

Windows 10

### Server operating system

Linux 2.6.32-954.3.5.lve1.4.89.el6.x86_64 x86_64

### Web server

Apache

### Database engine version

MariaDB

### PHP engine version

PHP 8.0

### Nextcloud version

24.0.2

### Updated from an older installed version or fresh install

Updated from an older version

### List of activated apps

```Text
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - announcementcenter: 6.3.1
  - appointments: 1.12.7
  - bookmarks: 10.5.1
  - bruteforcesettings: 2.4.0
  - calendar: 3.4.1
  - circles: 24.0.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.1.1
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - event_update_notification: 1.5.0
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - firstrunwizard: 2.13.0
  - geoblocker: 0.5.5
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - nextcloud_announcements: 1.13.0
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - passwords: 2022.6.20
  - photos: 1.6.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - recommendations: 1.3.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflowengine: 2.6.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_ldap
```

### Nextcloud configuration

```Text
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.2.1",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "PLAIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "maintenance": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "theme": "",
        "loglevel": 2,
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

**Comment by tcitworld** [comment]
The two are unrelated.
> The following appears in the Thunderbird error log
PUT https://nextcloud.redacted.xxx/index.php/apps/dashboard/
[HTTP/2 405 Method Not Allowed 290ms]

Is a misconfiguration.

**Comment by Digitwellsolutions** [comment]
Any clue where to look for the misconfiguration.

**Comment by tcitworld** [comment]
Either on web server configuration or Thunderbird side.

**Comment by pohutukawa** [comment]
I'm experiencing the same issue on Nextcloud 27.1.2 with Calendar 4.5.2, and I have for quite a while in the past with previous versions.

It would be helpful to find out any clues on **where** a configuration issue for this may need to be sought, as it sounds unlikely to be a web server or Thunderbird misconfiguration:

* The client sends a `.ics` file containing the location, so it should be just like any other field within the calendar invite. 
* I'm experiencing the same problem no matter whether I've tried to create a calendar event 
    * from Thunderbird,
    * from Evolution
    * or even the NC Calendar web front-end.

Any clues or insights that may help to get on top of this problem?

**Comment by Digitwellsolutions** [comment]
> I'm experiencing the same issue on Nextcloud 27.1.2 with Calendar 4.5.2, and I have for quite a while in the past with previous versions.
> 
> It would be helpful to find out any clues on **where** a configuration issue for this may need to be sought, as it sounds unlikely to be a web server or Thunderbird misconfiguration:
> 
>     * The client sends a `.ics` file containing the location, so it should be just like any other field within the calendar invite.
> 
>     * I'm experiencing the same problem no matter whether I've tried to create a calendar event
>       
>       * from Thunderbird,
>       * from Evolution
>       * or even the NC Calendar web front-end.
> 
> 
> Any clues or insights that may help to get on top of this problem?

The solution for me was related to security.
I had to add the following to my .htaccess
I can't remember exactly where I found this solution, but it works for me.
I understand it is probably overkill and I should be a bit more precise about the Mod_security thatgets turned off but I unfortunately don't have the skill or time to go any further.

<IfModule mod_security.c>
  SecRuleEngine Off
  SecFilterEngine Off
  SecFilterScanPOST Off
</IfModule>


<IfModule mod_security2.c>
  SecRuleEngine Off
  SecFilterEngine Off
  SecFilterScanPOST Off
</IfModule>

**Comment by Digitwellsolutions** [comment]
Seems I lost all of my formatting :-(
`<IfModule mod_security.c>
  SecRuleEngine Off
  SecFilterEngine Off
  SecFilterScanPOST Off
</IfModule>


<IfModule mod_security2.c>
  SecRuleEngine Off
  SecFilterEngine Off
  SecFilterScanPOST Off
</IfModule>`

**Comment by joshtrichards** [comment]
Yes, mod_security would definitely do it.  We document that here:

https://docs.nextcloud.com/server/latest/admin_manual/issues/general_troubleshooting.html#web-server-and-php-modules
